authentication/ldap_authentication_provider.py

```diff
@@ -12,35 +12,156 @@
 #  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  * See the License for the specific language governing permissions and
 #  * limitations under the License.
-
+import logging
 
 import ldap
 from flask_securest.authentication_providers.abstract_authentication_provider \
     import AbstractAuthenticationProvider
 from flask_securest import utils
+from flask_securest import constants
+from flask_securest.exceptions import AuthenticationException
+
+DN_ATTRIBUTE_NAME = 'DN'
+DEFAULT_AUTH_ATTR_NAME = 'uid'
+
+REQ_SEARCH_FIELDS = ('base_dn', 'admin_user_id', 'admin_password')
 
 
 class LDAPAuthenticationProvider(AbstractAuthenticationProvider):
-    def __init__(self, directory_url):
-        self.directory_url = directory_url
 
+    _logger = logging.getLogger(constants.FLASK_SECUREST_LOGGER_NAME)
+
+    def __init__(self, ldap_url, search_properties=None):
+        self.ldap_url = ldap_url
+        self.search_props = search_properties
+        self.validate_search_props()
+
+    def validate_search_props(self):
+        if self.search_props:
+            valid = all(test in self.search_props for test in
+                        REQ_SEARCH_FIELDS)
+            if not valid:
+                err = 'The configured search_properties are missing required '\
```

Comment:
1. This message has the same problem. An empty value will still cause the message to be something like:
"Search properties must hold values for the following fields: 'a, b, c', got 'a, b, c'

2. brackets of different types will appear in the message:
Search properties must hold values for the following fields: '('base_dn', 'admin_user_id', 'admin_password')', got '['admin_password', 'admin_user_id', 'base_dn']'